### PR TITLE
Decompose Release flow and introduce some fixes

### PR DESCRIPTION
**Pull request by OdNairy**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

Decompose Release flow into the smaller jobs that can be easily re-runed or fixed on-fly. PR also includes a massive of fixes for release flow to make it work correctly on actual tag push.

Screenshot 2022-01-20 at 17 36 37

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -91,6 +104,7 @@ workflows:
       - unit-test-sdk:
           name: "Run Unit tests"
           xcode: "12.5.0"
+      - build-tests-for-simulator
```
Comment (macdrevx) [inline .circleci/config.yml:107]: Why is this needed?

Comment (macdrevx) [reply .circleci/config.yml:107]: Ah, I see it's building the test host tests. Can you clarify that in the name. maybe `build-test-host-tests` or something like that

File: .circleci/config.yml [hunk 2]
```diff
@@ -347,8 +392,10 @@ jobs:
           name: Testing MapboxMaps with simulator
           command: |
             make test-sdk-without-building-simulator
```
Comment (macdrevx) [inline .circleci/config.yml:394]:
```suggestion
          command: make test-sdk-without-building-simulator
```

File: scripts/release/create-github-draft-release.sh [hunk 3]
```diff
@@ -7,51 +7,57 @@ set -euo pipefail
...
+    GL_NATIVE_RELEASE_URL=$(gh release view --repo mapbox/mapbox-gl-native-internal "maps-v$(jq -r .MapboxCoreMaps scripts/release/packager/versions.json)" --json url -q .url)
+    COMMON_RELEASE_URL=$(gh release view --repo mapbox/mapbox-sdk-common "v$(jq -r .MapboxCommon scripts/release/packager/versions.json)" --json url -q .url)
+
+    CHANGELOG=$( ([[ $(command -v parse-changelog) ]] && parse-changelog CHANGELOG.md) || echo "<Compose changelog here>" )
+
+    cat << EOF > notes.txt
+### Dependency requirements:
+
+* Compatible version of MapboxCoreMaps:
+* Compatible version of MapboxCommon:
```
Comment (macdrevx) [inline scripts/release/create-github-draft-release.sh:32]: Can we remove these too or auto-populate them?

File: scripts/release/create-github-draft-release.sh [hunk 4]
```diff
@@ -7,51 +7,57 @@ set -euo pipefail
...
+* Compatible version of Xcode:
+* Compatible version of macOS:
+
+### Changes
+
+$CHANGELOG
+
+### Dependencies
+- [GL Native]($GL_NATIVE_RELEASE_URL)
+- [Common]($COMMON_RELEASE_URL)
```
Comment (macdrevx) [inline scripts/release/create-github-draft-release.sh:42]: These links will only work for Mapbox employees. Can we remove them?

File: scripts/release/create-github-draft-release.sh [hunk 5]
```diff
@@ -7,51 +7,57 @@ set -euo pipefail
...
+### Dependency requirements:
+
+* Compatible version of MapboxCoreMaps:
+* Compatible version of MapboxCommon:
+* Compatible version of Xcode:
+* Compatible version of macOS:
+
+### Changes
+
+$CHANGELOG
```
Comment (macdrevx) [inline scripts/release/create-github-draft-release.sh:38]: This will be a good starting point, but there will be manual editing expected as well. CHANGELOG.md is meant to be very succinct, but we want release notes to be more descriptive.

File: scripts/release/create-github-draft-release.sh (hunk 4 above)
Comment (OdNairy) [reply scripts/release/create-github-draft-release.sh:42]: The point of these tickets is to get faster to the dependencies changelog. 
I didn't find the way to easily parse markdown from the shell to inject Public Changelog to the draft, but I suppose it should be fine to keep links to save some time looking for a correct release of Common or GL-native

File: .circleci/config.yml (hunk 2 above)
Comment (OdNairy) [reply .circleci/config.yml]: Done

File: scripts/release/create-github-draft-release.sh [hunk 6]
```diff
@@ -7,51 +7,57 @@ set -euo pipefail
...
+
+    CHANGELOG=$( ([[ $(command -v parse-changelog) ]] && parse-changelog CHANGELOG.md) || echo "<Compose changelog here>" )
+
+    cat << EOF > notes.txt
+### Dependency requirements:
+
+* Compatible version of MapboxCoreMaps:
+* Compatible version of MapboxCommon:
+* Compatible version of Xcode:
+* Compatible version of macOS:
```
[the remark this answers is not in the document]
Comment (OdNairy) [reply scripts/release/create-github-draft-release.sh]: Xcode and macOS version were removed.

File: scripts/release/create-github-draft-release.sh (hunk 3 above)
Comment (OdNairy) [reply scripts/release/create-github-draft-release.sh]: Added populating logic to the script

File: scripts/release/create-github-draft-release.sh (hunk 5 above)
Comment (OdNairy) [reply scripts/release/create-github-draft-release.sh:40]: Right, we still have to work manually with Release notes. The script purpose is to make changelog collection a bit easier

File: scripts/release/create-github-draft-release.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (macdrevx) [reply scripts/release/create-github-draft-release.sh]: Sorry, can we keep Xcode? I only meant to remove macOS.

[the remark this answers is not in the document]
Comment (OdNairy) [reply scripts/release/create-github-draft-release.sh]: Yeah. Should we hardcode some version here? Xcode 12.5.1?

File: scripts/release/create-github-draft-release.sh (hunk 4 above)
Comment (macdrevx) [reply scripts/release/create-github-draft-release.sh:46]: I think they should be removed before publishing the release notes though. I think it's frustrating for developers when they try to open links to internal stuff. Can you add some instructions for whoever is doing the release that they should replace these lines with the actual release notes from core and common?

File: scripts/release/create-github-draft-release.sh [hunk 7]
```diff
@@ -7,51 +7,61 @@ set -euo pipefail
...
+    MAPBOX_COREMAPS_VERSION=$(jq -r .MapboxCoreMaps "$VERSION_JSON_PATH")
+
+    CHANGELOG=$( ([[ $(command -v parse-changelog) ]] && parse-changelog CHANGELOG.md) || echo "<Compose changelog here>" )
+
+    cat << EOF > notes.txt
+### Dependency requirements:
+
+* Compatible version of MapboxCoreMaps: \`$MAPBOX_COREMAPS_VERSION\`
+* Compatible version of MapboxCommon: \`$MAPBOX_COMMON_VERSION\`
+* Compatible version of Xcode: \`12.5.1\`
```
Comment (macdrevx) [inline scripts/release/create-github-draft-release.sh:38]: This looks good

File: scripts/release/create-github-draft-release.sh (hunk 4 above)
Comment (OdNairy) [reply scripts/release/create-github-draft-release.sh:46]: I was going to add more info in Release ticket template from internal repo. Would you like to add me some info in place?

Comment (macdrevx) [reply scripts/release/create-github-draft-release.sh:46]: Your plan is fine. We can iterate if needed.

**Review by macdrevx — APPROVED** [review]
